src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -927,6 +955,23 @@ DeferErrorIfQueryNotSupported(Query *queryTree)
 		errorMessage = "subquery in OFFSET is not supported in multi-shard queries";
 	}
 
+	List *rteList = ExtractRangeTableEntryList(queryTree);
+	if (CheckRTEList(rteList, IsCitusLocalTableRTE))
+	{
+		preconditionsSatisfied = false;
+		errorMessage = "citus local tables cannot be used in distributed queries";
```

Comment:
err: Direct joins between local tables and distributed tables are not supported.
hint: Consider wrapping  %s to a CTE

Reply:
(for the hint, can you check the Citus source code and if we have a better wording, can you that? )